src/core/service.c

```diff
@@ -3093,6 +3093,8 @@ static void service_notify_message(Unit *u, pid_t pid, char **tags, FDSet *fds)
                         log_unit_warning(u, "Failed to parse MAINPID= field in notification message: %s", e);
                 else if (pid == s->control_pid)
                         log_unit_warning(u, "A control process cannot also be the main process");
+                else if (pid == getpid() || pid == 1)
```

Comment:
So `getpid()` is for the `systemd --user` case, and `1` for the "system" instance. Just wondering why that wouldn't  already be caught by the first condition?